src/NUnitFramework/tests/TestUtilities/TestBuilder.cs

```diff
@@ -135,6 +135,29 @@ public static ITestResult RunTestSuite(TestSuite suite, object testObject)
             return work.Result;
         }
 
+        public static CompositeWorkItem GenerateWorkItem(TestSuite suite, object testObject, ParallelScope? parallelScope)
+        {
+            TestExecutionContext context = new TestExecutionContext();
+            context.TestObject = testObject;
+            if(parallelScope != null) context.ParallelScope = (ParallelScope)parallelScope;
+
+            CompositeWorkItem work = (CompositeWorkItem)WorkItem.CreateWorkItem(suite, TestFilter.Empty);
+            work.InitializeContext(context);
+            work.Execute();
+
+            // TODO: Replace with an event - but not while method is static
+            while (work.State != WorkItemState.Complete)
+            {
+#if PORTABLE
+                System.Threading.Tasks.Task.Delay(1);
+#else
+                Thread.Sleep(1);
+#endif
+            }
+
+            return work;
+        }
```

Comment:
Did you find you actually had to execute the work items in order to check the order? I was guessing you would just call CreateChildWorkItems and then check without executing.